wgsl/index.bs

```diff
@@ -5231,6 +5231,80 @@ See [[#builtin-inputs-outputs]] for how to declare a built-in variable.
   </xmp>
 </div>
 
+# Operators # {#operators}
+
+[SHORTNAME] has symbol-based operators, such as the `+` in `a + b`.
+
+## Arithmatic operators ## {#arithmatic-operators}
+
+<table class='data'>
+  <caption>Binary operators</caption>
+  <thead>
+    <tr><th>Category<th>Preconditions<th>Conclusions
+  </thead>
+  <tr>
+    <td>Scalar arithmatic
+    <td>S: f32 or i32 or u32
+    <td>
+      <ul>
+        <li>`(S + S) -> S` (e.g. OpFAdd)
+        <li>`(S - S) -> S` (e.g. OpFSub)
+        <li>`(S * S) -> S` (e.g. OpFMul)
+        <li>`(S / S) -> S` (e.g. OpFDiv)
+        <li>`(S % S) -> S` (e.g. OpFMod)
+      </ul>
+  <tr>
+    <td>Vector arithmatic
+    <td>V: vecN<S>; S: f32 or i32 or u32
+    <td>
+      <ul>
+        <li>`(V + V) -> V` (component-wise, e.g. OpFAdd)
+        <li>`(V - V) -> V` (component-wise, e.g. OpFSub)
+        <li>`(V * V) -> V` (component-wise, e.g. OpFMul)
+        <li>`(V / V) -> V` (component-wise, e.g. OpFDiv)
+        <li>`(V % V) -> V` (component-wise, e.g. OpFMod)
+      </ul>
+  <tr>
+    <td>Vector OP Scalar "splats"
+    <td>V: vecN<S>; S: f32 or i32 or u32
+    <td>
+      <ul>
+        <li>`(V + S) -> (V    + V(S)) -> V`
+        <li>`(S + V) -> (V(S) + V   ) -> V`
+        <li>`(V - S) -> (V    - V(S)) -> V`
+        <li>`(S - V) -> (V(S) - V   ) -> V`
+        <li>`(V * S) -> (V    * V(S)) -> V` (e.g. OpVectorTimesScalar)
+        <li>`(S * V) -> (V(S) * V   ) -> V` (e.g. OpVectorTimesScalar)
+        <li>`(V / S) -> (V    / V(S)) -> V`
+        <li>`(S / V) -> (V(S) / V   ) -> V`
+        <li>`(V % S) -> (V    % V(S)) -> V`
+        <li>`(S % V) -> (V(S) % V   ) -> V`
+      </ul>
+  <tr>
+    <td>Matrix arithmatic
+    <td>S: f32
+    <td>
+      <ul>
+        <li>`(matAxB<S> + matAxB<S>) -> matAxB<S>` (component-wise)
+        <li>`(matAxB<S> - matAxB<S>) -> matAxB<S>` (component-wise)
+        <li>`(matAxB<S> * matCxA<S>) -> matCxB<S>` (linear-algebraic, e.g. OpMatrixTimesMatrix)
```

Comment:
Is this really right? Not `(matAxB<S> * matBxC<S>) -> matAxC<S>`?

Reply:
Our AxB are A columns and B rows, so columns on the left match the rows on the right, and we end up with `matCxB`. So the PR looks to be correct.

Reply:
Alternatively, matBxC * matAxB -> matAxC, which I think is probably a better way to phrase it. (You can see the right-to-left composition)